topostats/plotting.py

```diff
@@ -218,16 +225,17 @@ def sns_violinplot(self) -> None:
     #     self.save_plot(outfile)
     #     return fig, ax
 
-    def sns_melt_data(self) -> pd.DataFrame:
+    @staticmethod
+    def melt_data(df: pd.DataFrame, stat_to_summarize: str, var_to_label: dict) -> pd.DataFrame:
         """Melt a dataframe into long format for plotting with Seaborn."""
-        self.melted_data = pd.melt(
-            self.df.reset_index(), id_vars=[self.molecule_id, self.image_id], value_vars=self.stat_to_sum
-        )
-        self.melted_data["variable"] = self.melted_data["variable"].map(self.var_to_label)
-        self.melted_data.rename({"image": "Image"}, axis=1, inplace=True)
-        self.image_id = "Image"
+        melted_data = pd.melt(df.reset_index(), id_vars=["molecule_number", "basename"], value_vars=stat_to_summarize)
+        melted_data["variable"] = melted_data["variable"].map(var_to_label)
+        # melted_data.rename({"image": "Image"}, axis=1, inplace=True)
```

Comment:
Could remove this and the next line as they're now redundant.

Reply:
Thanks for spotting this, I knew I would forget to remove these 😆